apic_ml2/neutron/services/l3_router/apic_driver.py

```diff
@@ -0,0 +1,246 @@
+
+    def update_router_postcommit(self, context, router):
+        context._plugin = self
+        with apic_mapper.mapper_context(context) as ctx:
+            arouter_id = router['id'] and self.name_mapper.router(
+                ctx, router['id'], openstack_owner=router['tenant_id'])
+            tenant_id = self.aci_mech_driver._get_router_aci_tenant(router)
+
+        with self.manager.apic.transaction() as trs:
+            vrf = self.aci_mech_driver._get_tenant_vrf(router['tenant_id'])
+            # A Neutron router is rendered as a contract. This contract
+            # Will exist in the COMMON tenant when single tenant mode is false,
+            # in the sys_id tenant otherwise.
+            self.manager.create_router(arouter_id, owner=tenant_id,
+                                       transaction=trs,
+                                       context=vrf['aci_name'])
+            if router['admin_state_up']:
+                self.manager.enable_router(arouter_id, owner=tenant_id,
+                                           transaction=trs)
+            else:
+                self.manager.disable_router(arouter_id, owner=tenant_id,
+                                            transaction=trs)
+
+    def create_floatingip_postcommit(self, context, fip_context):
+        original = fip_context.original
+        floatingip = fip_context.current
+        port_id = original.get('floatingip', {}).get('port_id')
+        self._notify_port_update(port_id)
+        if floatingip:
+            floatingip['status'] = self._update_floatingip_status(
+                context, floatingip['id'])
+
+    def update_floatingip_precommit(self, context, id, fip_context):
+        port_id = self._get_port_mapped_to_floatingip(context, id)
+        fip_context.port_id_list = [port_id]
+
+    def update_floatingip_postcommit(self, context, id, fip_context):
+        floatingip = fip_context.original
+        port_id_list = fip_context.port_id_list or []
+        port_id_list.append(floatingip.get('floatingip', {}).get('port_id'))
+        for p in port_id_list:
+            self._notify_port_update(p)
+        status = self._update_floatingip_status(context, id)
+        if fip_context.current:
+            fip_context.current['status'] = status
+
+    def delete_floatingip_precommit(self, context, id, fip_context):
+        port_id_list = [self._get_port_mapped_to_floatingip(context, id)]
+        fip_context.port_id_list = port_id_list
+
+    def delete_floatingip_postcommit(self, context, id, fip_context):
+        self._notify_port_update(fip_context.port_id_list[0])
+
+    def _get_port_mapped_to_floatingip(self, context, fip_id):
+        try:
+            fip = self._plugin.get_floatingip(context, fip_id)
+            return fip.get('port_id')
+        except l3.FloatingIPNotFound:
+            pass
+        return None
+
+    def _notify_port_update(self, port_id):
+        l2 = mechanism_apic.APICMechanismDriver.get_driver_instance()
```

Comment:
I think you can just use aci_mech_driver here instead of calling get_driver_instance()